docs/design-v1.md

```diff
@@ -0,0 +1,106 @@
+# Caching Service Design Doc
+
+## Purpose
+
+Provide a generalized cache store that can be used from any other service, allowing users to avoid re-running expensive computation that has already been completed for the same dataset.
+
+## MVP Specifications
+
+**_Set a cache value_**
+
+To set a cache value, send a POST request with content-type "multipart/form-data" (to facilitate the file upload).
+
+The fields are:
+
+* `service` - required - name of the service
+* `token` - required - authentication token for the service
+* `params` - required - an arbitrary set of data used in the cache's namespace
+* `file` - required - multipart form-data file to cache
+* `expiration_days` - optional - total time in days until the cache's file gets deleted
+
+```
+curl -i -X POST -H "Content-Type: multipart/form-data"
+  -F file=@/home/u/bio.zip
+  -F service=MyService
+  -F token=xyz
+  -F params="{..arbitrary_json_data..}"
+  -F expiration_days=30
+  https://<caching_service_url>/v1/set
+```
+
+If the cache succeeds, you will get back an empty 200 response.
+
+If your auth token is invalid, you will receive a 403. Otherwise, there should be no errors.
```

Comment:
What happens if the record already exists for that set of params?

Reply:
Was thinking it would do a replacement.
We can just do a 200 with `{status: 'created'}` or `{status: 'replaced'}` in the response